prime-router/src/main/kotlin/history/azure/DeliveryFunction.kt

```diff
@@ -203,6 +207,31 @@ class DeliveryFunction(
         }
     }
 
+    /**
+     * Fetches the items that were contained in the passed in report ID walking up the report lineage if necessary
+     */
+    @FunctionName("getReportItemsV1")
+    fun getReportItems(
+        @HttpTrigger(
+            name = "getReportItemsV1",
+            methods = [HttpMethod.GET],
+            authLevel = AuthorizationLevel.ANONYMOUS,
+            route = "v1/report/{reportId}/items"
+        ) request: HttpRequestMessage<String?>,
+        @BindingName("reportId") reportId: UUID
+    ): HttpResponseMessage {
+        val claims = AuthenticatedClaims.authenticate(request)
+        if (claims == null || !claims.authorized(setOf("*.*.primeadmin"))) {
+            logger.warn("User '${claims?.userName}' FAILED authorized for endpoint ${request.uri}")
+            return HttpUtilities.unauthorizedResponse(request, authenticationFailure)
+        }
+
+        val reportGraph = ReportGraph(workflowEngine.db)
+        val metadata = reportGraph.getMetadataForReports(listOf(reportId))
```

Comment:
This endpoint is here to provide an example for how the underlying CTEs would fit together.